### PR TITLE
daml-lf: interned strings/names in scenarios

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
We interned names/strings in protos for scenarios  

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 1]
```diff
@@ -48,18 +48,27 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
             onlySerializableDataDefs).decode))
   }
 
-  type ProtoModule = PLF.Module
+  type ProtoModule = PLF.Package
```
Comment (gerolf-da) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: Rename `ProtoModule` to `ProtoPackage`.

File: compiler/scenario-service/protos/scenario_service.proto [hunk 2]
```diff
@@ -327,10 +327,9 @@ message Map {
 // Modules that can be loaded. The actual message types are not
 // used here as LF modules can have deep nesting depth and it is
 // not easy to change the recursion depth limit in gRPC.
-message Module {
+message ScenarioModule {
   oneof module {
```
Comment (cocreature) [inline compiler/scenario-service/protos/scenario_service.proto]: What’s the point of having a `oneof` with only one entry?

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 3]
```diff
@@ -48,18 +48,37 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
             onlySerializableDataDefs).decode))
   }
 
-  type ProtoModule = PLF.Module
+  // LF scenario modules are wrapped in a distinct proto package
```
Comment (cocreature) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: This line is duplicated.

**Comment by gerolf-da** [comment]
I agree with @cocreature on including benchmarks in the decision.

**Pull request merged**